contracts/Tokenator.sol

```diff
@@ -0,0 +1,176 @@
+        );
+    }
+
+    /// @notice allows user to claim underlying token from this contract by burning tokenator token
+    /// @param tokenatorAmount amount of tokenator token to burn
+    /// @param to address to receive the underlying token
+    /// @return underlyingTokenAmount amount of underlyingToken `to` address receives
+    function claimUnderlyingToken(
+        uint256 tokenatorAmount,
+        address to
+    ) external returns (uint256 underlyingTokenAmount) {
+        if (commencement >= block.timestamp) revert CommencementNotPassed();
+        if (to == address(0)) revert ZeroAddress();
+
+        underlyingTokenAmount = underlyingTokenForTokenator(tokenatorAmount);
+        _burn(msg.sender, tokenatorAmount);
+        underlyingToken.safeTransfer(to, underlyingTokenAmount);
+
+        emit ClaimUnderlyingToken(
+            msg.sender,
+            to,
+            tokenatorAmount,
+            underlyingTokenAmount
+        );
+    }
+
+    /// @notice owner may withdraw underlying token in emergency situations
+    /// @param amount underlying token amount to withdraw
+    /// @param to address to receive the underlying token
+    function emergencyWithdraw(uint256 amount, address to) external onlyOwner {
+        if (to == address(0)) revert ZeroAddress();
+
+        underlyingToken.safeTransfer(to, amount);
+
+        emit EmergencyWithdrawal(msg.sender, amount, to);
+    }
+
+    /// @notice owner may withdraw tokens accidentally sent to the contract
+    /// @notice this function can't be used to withdraw the underlying token
+    /// @param token the token to withdraw
+    /// @param amount the amount of the token to withdraw
+    /// @param to the address to receive the rescued token
+    function rescueTokens(
+        IERC20 token,
+        uint256 amount,
+        address to
+    ) external onlyOwner {
+        if (to == address(0)) revert ZeroAddress();
+        if (token == underlyingToken) revert CannotRescueUnderlyingToken();
+
+        token.safeTransfer(to, amount);
+
+        emit TokensRescued(msg.sender, token, amount, to);
+    }
+
+    /// @notice returns whether the duration period is complete
+    /// @return isComplete true if the duration is complete, otherwise false
+    function complete() public view returns (bool isComplete) {
+        isComplete = ageSeconds() >= durationSeconds;
+    }
+
+    /// @notice returns the number of seconds elapsed from the commencement time
+    /// @return elapsedSeconds seconds elapsed from the commencement time
+    function ageSeconds() public view returns (uint256 elapsedSeconds) {
+        if (block.timestamp <= commencement) return 0;
+        elapsedSeconds = block.timestamp - commencement;
+    }
+
+    /// @notice returns the tokenator redeemable percent based on current time within duration
+    /// @notice scaled x18 => 1e18 = 1.00 = 100%
+    /// @return p18 redeemable percent x18
+    function redeemablePercent() public view returns (uint256 p18) {
```

Comment:
`redeemable<Percent>` may be misleading in terms of naming as it's not a percentage really? as it returns p18 which is a U18 FP integer like `underlyingTokenPerTokenator`, so maybe something like `redeemableFraction`?

Reply:
I think redeemablePercent is fine and gets the message across. redeemableFraction would need the same kind of explanation

Reply:
yeah `redeemableFraction` feels more confusing to me than `redeemablePercent`